Object types page rework (#77)
documaster/noark5-web-services: version1/02 - Object types.md

The following notation is used in the tables below:

- **V**
    - denotes **true** (a checkmark)

> - **V**

Reviewer: I guess you meant the question mark here, although I'm not completely sure if we need that separate notation now that you have separated the Finalizable Object fields. The question mark is only used there for the `avsluttetAv` and `avsluttetAvBrukerIdent` and I think the relation there could be explained in the description for the Finalizable Object itself. What do you think?
```suggestion
- **?**
```

Author: Actually the question mark here is a miss. It should be blank as the fields can be NULL.

Reviewer: Removed with 098fcd235b6241f72af2f2824df4757cd83a35a9

>     - denotes **conditional**, contidition(s) are specified in the "Default" or "Comment" columns

Reviewer: This is still confusing and I think we can drop it so that the checkmark only denotes true?

Author: Removed. It was a leftover from the "?" which is somehow missed :(

- **Unlink**
    - This field can be used in Unlink operations. (ref* fields)

### User permissions
**Permissions** - required user rights/permissions to carry on update operations
- **USM**
    - Update System Managed - allows modification of special system fields
- **EFO**

>     - Edit Finalized Objects - allows modification of finalized objects

Reviewer 2: It would be more correct if it says that this applies to finalized objects and _children of finalized objects_.

I would also describe what finalized objects are as this may not be entirely obvious to someone new to Noark.

This partially applies to "journaled" below, but I will write a separate comment for that.

```suggestion
    - Edit Finalized Objects
        - allows modification of finalized objects or children of finalized objects
        - a finalized object is an object whose avsluttetDato is set
```

- **EJO**

>     - Edit Journaled Objects - allows modification of journaled objects

Reviewer 2: You can also add two more permissions:
* publishing (you refer to it further down)
* journal objects

Sample text:
* JO - Journal objects
  * allows the journaling of registry entries (journalpost)
  * "journalling" implies setting the `journalstatus` to J
* PO - Publish objects
  * allows the publishing and unpublishing of records
  * "publishing" implies setting the `publiseringsstatus` or `eInnsynStatus` of any record type (registrering, arkivnotat, moeteregistrering, journalpost)

## Archives and archive sections

### Basic Archival Object

The **Basic Archival Object** contains the common fields inherited by all archival objects in the system.

> Note: **EFO** required if finalized

Reviewer 2: For all practical purposes, EFO applies to all entities.

As such, I'd recommend that this is extracted to its own section instead of being repeated on all objects. This will also allow you to add more information about what "finalization" actually means in the context of Noark - a non-null finalized date and a finalized status (for entities that have statuses).

| Field no. | Type      | Field                  | Not null |     Default     | Code list      |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:-----------------------|:--------:|:---------------:|:---------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|

> |           | string    | id                     |    V     | Auto inc        |                |  |  sm  |  V   |   V   |  V   |      |        |         |

Reviewer 2: The id is an internal detail and its logic is also an internal detail. We cannot state that it is auto-incremented, because that may not be true in the future and is not true even now for some objects. I suggest using a different statement or leaving the default column empty. This is an "sm", so no integration is ever expected to have to bother itself with the logic of the default.

Mostly the same applies to "version" below.

Author: Changed to "Auto generated" without specifying how as leaving the default empty might be understood as not having such.

|           | string    | version                |    V     | Auto inc        |                |  |  sm  |  V   |   V   |      |      |        |         |
|           |           |                        |          |                 |                |  |      |      |       |      |      |        |         |

> | M001      | string    | uuid                   |    V     | Random uuid     |                |  |  cw  |  V   |   V   |  V   |      |        | USM     |

Reviewer 2: I'd also add a comment that it is strongly discouraged to manually set uuid. In fact, that applies to uuid and the opprettet fields below.

| M600      | timestamp | opprettetDato          |    V     | Now             |                |  |  cw  |  V   |   V   |  V   |      |        | USM     |
| M601      | string    | opprettetAv            |    V     | Current user    |                |  |  cw  |  V   |   V   |  V   |      |        | USM     |

> | M601      | string    | opprettetAvBrukerIdent |    V     | Current user ID |                |  |  cw  |  V   |   V   |  V   |      |        | USM     |

Reviewer 2: We can expand on the `opprettet*` field defaults the way you have for some other fields - journalansvarlig/saksansvarlig. Namely, the following is true;
* if any of the opprettetAv* fields is provided, both have to be provided
* if none is provided, Documaster will default to the current user.

Note: **EFO** required if finalized
| Field no. | Type      | Field                  | Not null |              Default             | Code list      |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                      |
|:----------|:----------|:-----------------------|:--------:|:--------------------------------:|:---------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:-----------------------------------------------------------------------------------------------------------------------------|
| M602      | timestamp | avsluttetDato          |          | Now                              |                |  |  cw  |  V   |   V   |  V   |      |        | <p>&bull; If finalized M602 and both M603 must not be null</p><p>&bull; If not finalized M602 and both M603 must be null</p> |
| M603      | string    | avsluttetAv            |          | CurentUser if both M603 are null |                |  |  cw  |  V   |   V   |  V   |      |        | <p>&bull; If finalized M602 and both M603 must not be null</p><p>&bull; If not finalized M602 and both M603 must be null</p> |

> | M603      | string    | avsluttetAvBrukerIdent |          | CurentUser if both M603 are null |                |  |  cw  |  V   |   V   |  V   |      |        | <p>&bull; If finalized M602 and both M603 must not be null</p><p>&bull; If not finalized M602 and both M603 must be null</p> |

Reviewer 2: I think these comments need to be improved to enhance clarity.

1. I'd add a comma after _If finalized_, otherwise the whole sentence reads as a condition where as `If finalized, M602 and...` reads differently.
2. For entities that have statuses (files, records, documents), the integration is not required to set avsluttetDato - it will default to "now". The comment currently states that it is required, but that is only true for non-status entities.

If you decide to follow my other comments to extract additional information about what finalization means, then I'd recommend no other changes. But, if you do not, it would be worth stating what "finalizing" means:
* setting the finalized date for entities that do not have statuses; or
* setting the status for entities that have statuses

Note: **EFO** required if finalized
| Field no. | Type      | Field                  | Not null |   Default   | Code list |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:-----------------------|:--------:|:-----------:|:----------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
| M006      | string    | arkivskaperIdent       |    V     |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
| M023      | string    | arkivskaperNavn        |    V     |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
| M021      | string    | beskrivelse            |          |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
|           |           |                        |          |             |           |  |      |      |       |      |      |        |         |
|           | ref       | refArkiv               |    V     |             |           |  |      |  V   |   V   |      |  V   |        | &nbsp;  |

### Arkivdel ([Finalizable Object](#finalizable-object-basic-archival-object))
(Series)

Note: **EFO** required if finalized

> Note: *Arkivdel* can have either subfolders or records but not both

Reviewer 2: It's technically not a subfolder in the context of a series. It's just a folder. 

The beginning of the document already states that files are bundled under the term "AbstraktMappe" and records under the term "AbstraktRegistrering", so I think you can use the same terms to avoid confusion and questions whether "subfolders" also implies case files, etc. See suggestion below

Reviewer 2: ```suggestion
Note: *Arkivdel* can have either AbstraktMappe objects (saksmappe, mappe, moetemappe) or some AbstraktRegistrering objects (basisregistrering, arkivnotat), but not both at the same time.
```

This also additionally indicates that only some record types are supported in series.

| M021      | string    | beskrivelse                       |          |             |                |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                               |
| M051      | string    | arkivdelstatus                    |    V     |      V      | arkivdelstatus |  |  V   |  V   |   V   |      |      |        |                                                                                                                                               |
| M300      | string    | dokumentmedium                    |    V     |      V      | dokumentmedium |  |  V   |  V   |   V   |      |      |        |                                                                                                                                               |
| M107      | date      | arkivperiodeStartDato             |    V     | Now         |                |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                               |

> | M108      | date      | arkivperiodeSluttDato             |          |             |                |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; Mandatory if arkivdelstatus is set to overlapping period</p>                                                                        |

Reviewer 2: You may indicate that the "overlap period" is `O` in the comment (I doubt many integrators read Noark 😅 )

|           | date      | overlappingsperiodeSluttdato      |          |             |                |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                               |
| M500      | string    | skjerming                         |          |             | skjerming      |  |  V   |  V   |   V   |      |      |        |                                                                                                                                               |
|           |           |                                   |          |             |                |  |      |      |       |      |      |        |                                                                                                                                               |
|           | ref       | refArkiv                          |    V     |             |                |  |      |  V   |   V   |      |  V   |   V    |                                                                                                                                               |

> |           | ref       | refPrimaerKlassifikasjonssystem   |          |             |                |  |      |  V   |   V   |      |  V   |   V    | <p>Mandatory:</p><p>&bull; If null inherited from a child</p><p>&bull; If not null all childs must have the same primary KS </p>;             |

Reviewer 2: I'd edit the comment to enhance its clarity.
1. The primary classification is not mandatory (it is nullable)

As with other comments, you are missing a comma after `If null`, but I think the sentence can be improved overall:
```
If not null, children will be required to specify a primary class. If null, children must not specify a primary class.
```

|           | ref       | refSekundaerKlassifikasjonssystem |          |             |                |  |      |      |   V   |      |  V   |   V    |                                                                                                                                               |
|           | ref       | refMappe                          |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktMappe**</p>                                                                                                    |
|           | ref       | refRegistrering                   |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Basisregistrering**</p>                                                                                                |
|           | ref       | refArkivdelForloeper              |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p>                                                                       |

> |           | ref       | refArkivdelArvtaker               |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p><p>&bull; Mandatory if arkivdelstatus is set to overlapping period</p> |

Reviewer 2: I believe you meant "self references"? (cross references has a different meaning in the context of noark)

|           |           |                               |          |             |                   |  |      |      |       |      |      |        |                                                                                                                                                                    |
|           | ref       | refKlassifikasjonssystem      |    V     |             |                   |  |      |  V   |   V   |      |  V   |   V    |                                                                                                                                                                    |
|           | ref       | refForelderKlasse             |          |             |                   |  |      |  V   |   V   |      |  V   |   V    |                                                                                                                                                                    |

> |           | ref       | refBarnKlasse                 |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; Cannot have both refBarnKlasse and kassasjonsvedtak</p>                                                                                                  |

Reviewer 2: I think you can add the same comment to the kassasjonsvedtak row for clarity.

|           | ref       | refMappeSomSekundaer          |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktMappe** as a secondary **Klasse**</p> <p>&bull; **refBarnKlasse** cannot be set if **refMappeSomSekundaer** is set</p>              |
|           | ref       | refRegistreringSomPrimaer     |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktRegistrering** as a primary **Klasse**</p> <p>&bull; **refBarnKlasse** cannot be set if **refRegistreringSomPrimaer** is set</p>    |
|           | ref       | refRegistreringSomSekundaer   |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktRegistrering** as a secondary **Klasse**</p><p>&bull; **refBarnKlasse** cannot be set if **refRegistreringSomSekundaer** is set</p> |

## Folders

> ### Abstract mappe ([Finalizable Object](#finalizable-object-basic-archival-object))

Reviewer 2: You can specify that this is the base mappe object type (the way you have done for the other inheritable objects)

Note: **EFO** required if finalized
| Field no. | Type      | Field                            | Not null |     Default     | Code list         |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                         |
|:----------|:----------|:---------------------------------|:--------:|:---------------:|:------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------------------------------------------------------------------------------------------------------------------------------|

> | M003      | string    | mappeIdent                       |    V     | Auto set to ID  |                   |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                 |

Reviewer 2: I'd state that the default is different for some file types in the comments section and urge them to look at the other tables for details (as you have described it correctly in the saksmappe table)

|           |           |                                  |          |                 |                   |  |      |      |       |      |      |        |                                                                                                                                 |
| M711      | object    | virksomhetsspesifikkeMetadata    |          |                 |                   |  |  V   |  V   |       |      |      |        |                                                                                                                                 |
|           |           |                                  |          |                 |                   |  |      |      |       |      |      |        |                                                                                                                                 |
|           | ref       | refArkivdel                      |    c     |                 |                   |  |      |  V   |   V   |      |  V   |   V    | <p>&bull; exactly one of **refArkivdel** and **refForelderMappe** must be set</p>                                               |

> |           | ref       | refRegistrering                  |          |                 |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Basisregistrering**</p><p>&bull; exactly one of **refBarnMappe** and **refRegistrering** must be set</p> |

Reviewer 2: The comment no longer holds for the base type. I'd state in the comment that different files support different record types and urge readers to look at the other tables for details. You can then describe those details in the other tables:
* mappe: only supports basisregistrering
* saksmappe: supports all types, but moeteregistrering
* moetemappe: only supports moeteregistrering

|           |           |                                  |          |             |                   |  |      |      |       |      |      |        |                                                                                                                                                                           |
|           | ref       | refForelderMappe                 |    c     |             |                   |  |      |  V   |   V   |      |  V   |   V    | <p>&bull; references a parent **Mappe**</p><p>&bull; exactly one of **refArkivdel** and **refForelderMappe** must be set</p><p>&bull; Nesting limited to 5 max levels</p> |
|           | ref       | refBarnMappe                     |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references children **Mappe**</p><p>&bull; exactly one of **refBarnMappe** and **refRegistrering** must be set</p><p>&bull; Nesting limited to 5 max levels</p> |

### Saksmappe ([Finalizable Object](#finalizable-object-basic-archival-object)) 

> (Abstract Folder)

Reviewer 2: ```suggestion
(Case file)
```

Reviewer: This comment is not addressed.

| M025      | string    | offentligTittel                  |          |                 |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                      |
| M021      | string    | beskrivelse                      |          |                 |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                      |

> | M300      | string    | dokumentmedium                   |    V     |      V          | dokumentmedium     |  |  V   |  V   |   V   |      |      |        |                                                                                                                                                                                                                                                                                                      |

Reviewer 2: Fields tittel, offentligtittel, beskrivelse, dokumentmedium, kassasjonsvedtak, kassasjonshlemmel, beverangstid, skjerming, forventetKassasjonsDato, BSM, and most of the ref fields are inherited from the base file described elsewhere. As a result, they can be removed from this table.

## Records

### Basisregistrering ([Finalizable Object](#finalizable-object-basic-archival-object))
(Basic Record)

**Basisregistrering** introduces the common fields for all *registrering* objects.

Note: **EFO** required if finalized

| Field no. | Type      | Field                            | Not null |    Default    | Code list          |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                                                                                                                                                                  |
|:----------|:----------|:---------------------------------|:--------:|:-------------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | M004      | string    | registreringsIdent               |    V     | Auto inc      |                    |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                          |

Reviewer 2: As with the file type, I would not say that it is auto-incremented. I'd say that it is the `id` for some types and record year/record sequence number for others. We can describe a general default here and state that it may be overridden in children (as is the case with registry entry)

| M024      | string    | forfatter                        |          |               |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                          |
|           | string    | publiseringsstatus               |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED, PUBLISHED_METADATA, PUBLISHED_METADATA_AND_DOCUMENTS. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                             |

> |           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                                                                    |

Reviewer 2: I'd use the actual value for the default (to avoid confusion) - NOT_PUBLISHED

> |           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                                                                    |

Reviewer 2: The comment about _PublishPermission_ is not correct.

When unpublishing (NOT_PUBLISHED), the permission is still required. Maybe you were misled by the fact that we allow a user to create a record in state NOT_PUBLISHED and then, yes, we would not require the permission.

In other words, I would state that modifying this field requires the _publish permission_ (and not limit the documentation to "not NOT_PUBLISHED"

> |           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                                                                    |

Reviewer 2: We support more states for eInnsyn publishing since 3.11.0 that we may add (ref unpublishing)

Author: I also removed the comment about 3.6.1 as the document is supposed to reflect the latest version.

| M085      | string    | moeteregistreringstype           |    V     |                    | moeteregistreringstype   |  |  V   |  V   |   V   |      |      |        |         |
| M088      | string    | moetesakstype                    |          |                    | moetesakstype            |  |  V   |  V   |   V   |      |      |        |         |
| M055      | string    | moeteregistreringsstatus         |          |                    | moeteregistreringsstatus |  |  V   |  V   |   V   |      |      |        |         |

### Arkivnotat ([Basisregistrering](#basisregistrering-finalizable-object))
(Archive Note)

_Since: 3.6.1_

Note: **EFO** required if finalized

Note: **EJO** required if journaled
| Field no. | Type      | Field                            | Not null |   Default   | Code list          |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:---------------------------------|:--------:|:-----------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|

> | M110      | date      | offentlighetsvurdertDato         |          |             |                    |  |  V   |  V   |   V   |  V   |      |        |         |

Reviewer 2: This field actually belongs to the base type (basisregistrering in this documentation) and has to be moved.

Arkivnotat has support for `refArkivnotatDokumentflyt` that needs to be listed here, instead.

| M015      | number    | journalpostnummer                 |    V     |      V      |                    |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |
|           | string    | *journalansvarlig                 |    V     |      V      |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The name of the person responsible for this Journalpost (similar to M306 Saksmappe#saksansvarlig)</p>                                                                                                                                                                                                                                                                                                 |

> |           | string    | *journalansvarligBrukerIdent      |    V     |      V      |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The user ID of the person responsible for this Journalpost as stored in journalansvarlig (similar to M306 Saksmappe#saksansvarligBrukerIdent)</p>                                                                                                                                                                                                                                                     |

Reviewer 2: There are extra asterisks that have to be removed in the field names of journalansvarlig and journalansvarligBrikerIdent

Author: The asterisk is tehre because of the note below the table. Actually those fields are inherited but the constraints for them are extended. 
But, if you think the note under the table is enough or the asterisks are confusing I'm fine to remove them.

Reviewer 2: I think they are enough. In fact, the asterisks can stay, but my concern is that this is a markdown file after all and asterisk pairs have a special meaning. So, it would be fairly easy to mess up the formatting with such single asterisks.

Author: I considered replacing the asterisks with question marks but it just looks bad. So, unless switching to &ast; sound better, I'll leave them as they are.

| M082      | string    | journalposttype                   |    V     |             | journalposttype    |  |  V   |  V   |   V   |      |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |
| M053      | string    | journalstatus                     |    V     |      V      |  journalstatus     |  |  V   |  V   |   V   |      |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | M110      | date      | offentlighetsvurdertDato          |          |             |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |

Reviewer 2: This field belongs to basisregistrering (see other comments), so it can be removed from here.

| M051      | string    | arkivdelstatus                    |    V     |      V      | arkivdelstatus |  |  V   |  V   |   V   |      |      |        |                                                                                                                                               |
| M300      | string    | dokumentmedium                    |    V     |      V      | dokumentmedium |  |  V   |  V   |   V   |      |      |        |                                                                                                                                               |
| M107      | date      | arkivperiodeStartDato             |    V     | Now         |                |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                               |
| M108      | date      | arkivperiodeSluttDato             |          |             |                |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; Mandatory if arkivdelstatus is set to overlapping period</p>                                                                        |

> |           | date      | overlappingsperiodeSluttdato      |          |             |                |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                               |

Reviewer: This date field (`overlappingsperiodeSluttdato`) is also mandatory when arkivdelstatus is set to overlapping period.

| M088      | string    | moetesakstype                    |          |                    | moetesakstype            |  |  V   |  V   |   V   |      |      |        |         |
| M055      | string    | moeteregistreringsstatus         |          |                    | moeteregistreringsstatus |  |  V   |  V   |   V   |      |      |        |         |

### Arkivnotat ([Basisregistrering](#basisregistrering-finalizable-object))
(Archive Note)

_Since: 3.6.1_

Note: **EFO** required if finalized

Note: **EJO** required if journaled
| Field no. | Type      | Field                            | Not null |   Default   | Code list          |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:---------------------------------|:--------:|:-----------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
|           |           | refArkivnotatDokumentflyt        |          |             |                    |  |      |      |   V   |      |  V   |   V    |         |

> ### Journalpost ([Registry Entry](#registry-entry-basisregistrering))

Reviewer: Should be Basisregistrering as is with the other record types:
```suggestion
### Journalpost ([Basisregistrering](#registry-entry-basisregistrering))
```

> ### Journalpost ([Basisregistrering](#registry-entry-basisregistrering))

Reviewer: To fix the anchor link:
```suggestion
### Journalpost ([Basisregistrering](#basisregistrering-finalizable-object))
```

| M660      | string    | flytTilBrukerIdent     |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M661      | timestamp | flytMottattDato        |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M662      | timestamp | flytSendtDato          |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M664      | string    | flytMerknad            |          |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M663      | string    | flytStatus             |    V     |             | flytStatus |  |  V   |  V   |   V   |      |      |        |                                                                     |

> |           |           |                        |          |             |            |  |      |      |       |      |      |        |                                                                     |

Reviewer: Needs a link to the Base Object:
```suggestion
### Merknad ([Basic Archival Object](#basic-archival-object))
```

- **Unlink**
    - This field can be used in Unlink operations. (ref* fields)

### User permissions

> **Permissions** - required user rigts/permissions to carry on update operations

Reviewer: Typo `rigts` -> `rights`
```suggestion
**Permissions** - required user rights/permissions to carry on update operations
```

|           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                                                                    |
| M300      | string    | dokumentmedium                   |    V     |      V        | dokumentmedium     |  |  V   |  V   |   V   |      |      |        |                                                                                                                                                                                                                                                                          |

> |           | string    | journalansvarlig                 |          |               |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The name of the person responsible for this Basisregistrering (similar to M306 Saksmappe#saksansvarlig)</p><p>&bull; Both journalansvarlig and journalansvarligBrukerIdent must be null or specified<p>&bull; </p>                                             |

Reviewer: Extra paragraph tag and bullet point not needed:
```suggestion
|           | string    | journalansvarlig                 |          |               |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The name of the person responsible for this Basisregistrering (similar to M306 Saksmappe#saksansvarlig)</p><p>&bull; Both journalansvarlig and journalansvarligBrukerIdent must be null or specified</p>                                             |
```

> |           | string    | journalansvarligBrukerIdent      |          |               |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The user ID of the person responsible for this Basisregistrering as stored in journalansvarlig (similar to M306 Saksmappe#saksansvarligBrukerIdent)</p><p>&bull; Both journalansvarlig and journalansvarligBrukerIdent must be null or specified<p>&bull; </p> |

Reviewer: Extra paragraph tag and bullet point not needed:
```suggestion
|           | string    | journalansvarligBrukerIdent      |          |               |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The user ID of the person responsible for this Basisregistrering as stored in journalansvarlig (similar to M306 Saksmappe#saksansvarligBrukerIdent)</p><p>&bull; Both journalansvarlig and journalansvarligBrukerIdent must be null or specified</p> |
```

- **Unlink**
    - This field can be used in Unlink operations. (ref* fields)

### User permissions
**Permissions** - required user rigts/permissions to carry on update operations
- **USM**
    - Update System Managed - allows modification of special system fields
- **EFO**
    - Edit Finalized Objects - allows modification of finalized objects
- **EJO**

>     - Edit Journaled Objects - allows modification of journaled objects

Reviewer: Only Registry Entries (Journalpost) can be journaled and require that permission.

Author: This is just description of the abbreviation used later in the document.

Reviewer: Yes, but the permission is listed as requred under all record types. It is not needed/applicable for basis registrering, arkivnotat and moeteregistrering.

Reviewer 2: I'd also describe what "journaled objects" means - journalstatus of registry entries to be J. You can add bullet points to the permissions as this will lead to more information having to be added to the same line. A related comment has been posted for EFO above.

|           | ref       | refPrimaerKlassifikasjonssystem   |          |             |                |  |      |  V   |   V   |      |  V   |   V    | <p>Mandatory:</p><p>&bull; If null inherited from a child</p><p>&bull; If not null all childs must have the same primary KS </p>;                |
|           | ref       | refSekundaerKlassifikasjonssystem |          |             |                |  |      |      |   V   |      |  V   |   V    |                                                                                                                                                  |
|           | ref       | refMappe                          |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktMappe**</p>                                                                                                       |
|           | ref       | refRegistrering                   |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Basisregistrering**</p>                                                                                                   |

> |           | ref       | refArkivdelForloeper              |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p><p>&bull</p> |

Reviewer: No need for the empty second paragraph + bullet point:
```suggestion
|           | ref       | refArkivdelForloeper              |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p> |
```

|           | ref       | refRegistreringSomSekundaer   |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktRegistrering** as a secondary **Klasse**</p><p>&bull; **refBarnKlasse** cannot be set if **refRegistreringSomSekundaer** is set</p> |

## Folders

### Abstract mappe ([Finalizable Object](#finalizable-object-basic-archival-object))

Note: **EFO** required if finalized
| Field no. | Type      | Field                            | Not null |     Default     | Code list         |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                         |
|:----------|:----------|:---------------------------------|:--------:|:---------------:|:------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------------------------------------------------------------------------------------------------------------------------------|

> | M003      | string    | mappeIdent                       |          | Auto set to ID  |                   |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                 |

Reviewer: `mappeIdent` cannot be null, you have accidentally removed the check mark.

(Folder)

Note: **EFO** required if finalized
| Field no. | Type      | Field                            | Not null |   Default   | Code list         |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                         |
|:----------|:----------|:---------------------------------|:--------:|:-----------:|:------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------------------------------------------------------------------------------------------------------------------------------|
|           | string    | mappestatus                      |          |             | mappestatus       |  |  V   |  V   |   V   |      |      |        | _Since 3.6.1_                                                                                                                   |
|           |           |                                  |          |             |                   |  |      |      |       |      |      |        |                                                                                                                                 |

> |           | ref       | refForelderMappe                 |    c     |             |                   |  |      |  V   |   V   |      |  V   |   V    | <p>&bull; references a parent **Mappe**</p><p>&bull; exactly one of **refArkivdel** and **refForelderMappe** must be set</p><p>&bull; Nesting limited to 5 max levels</p>    |

Reviewer: The comment for the nesting up to 5 levels maximum can possibly be moved to the `refBarnMappe` field reference. That's more of a suggestion from my side, not something I insist on 😄

Author: Lets have it on both :)

|           | ref       | refBarnMappe                     |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references children **Mappe**</p><p>&bull; exactly one of **refBarnMappe** and **refRegistrering** must be set</p>    |

> ### Saks mappe ([Finalizable Object](#finalizable-object-basic-archival-object)) 

Reviewer: Saksmappe should be one word:
```suggestion
### Saksmappe ([Finalizable Object](#finalizable-object-basic-archival-object)) 
```

|:----------|:----------|:---------------------------------|:--------:|:---------------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|           | string    | prefiks                          |          |                 |                    |  |  cw  |  V   |   V   |  V   |      |        | <p>&bull; optional identifier of the system this **Saksmappe** originated from</p><p>&bull; it is strongly advised to populate this field when the target system contains or will contain data from multiple source case management systems</p><p>&bull; once set, this value cannot be modified</p> |

> | M003      | string    | mappeIdent                       |    V     | Case Year/ID    |                    |  |  cw  |  V   |   V   |  V   |      |        | Default set to Case Year/Case sequence ID, i.e. "2025/13"                                                                                                                                                                                                                                            |

Reviewer: The mappeIdent for Case Files is `case_year/case_sequence_number`:
```suggestion
| M003      | string    | mappeIdent                       |    V     | Case Year/Case Sequence Number    |                    |  |  cw  |  V   |   V   |  V   |      |        | Default set to Case Year/Case sequence ID, i.e. "2025/13"                                                                                                                                                                                                                                            |
```

Author: It is described in the Comment column. I've put just "ID" in the Default column to avoid unnecessary widening of the column. I'm also fine to update the Default if you think the comment is not enough.

| M100      | date      | saksdato                         |    V     | Now             |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                      |
| M305      | string    | administrativEnhet               |    V     |                 | administrativEnhet |  |  V   |  V   |   V   |      |      |        |                                                                                                                                                                                                                                                                                                      |

> | M306      | string    | saksansvarlig                    |    V     | Current user    |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; if both saksansvarlig and saksansvarligBrukerIdent are null default applies.</p><p>&bull; Not allowed to have only one of saksansvarlig or saksansvarligBrukerIdent as null</p>                                                                                                            |

Reviewer: The wording is a bit confusing since we have explicitly listed both fields as not null, which is correct, so we need to make a better distinction between preserved as not null and supplied as null:
```suggestion
| M306      | string    | saksansvarlig                    |    V     | Current user    |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; if both saksansvarlig and saksansvarligBrukerIdent are left as null, the defaults apply.</p><p>&bull; Not allowed to supply only one of saksansvarlig or saksansvarligBrukerIdent and leave the other as null</p>                                                                                                            |
```

Author: I had a hard time finding the right words, so thanks :)

| M021      | string    | beskrivelse                      |          |               |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                          |
| M024      | string    | forfatter                        |          |               |                    |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                          |

> |           | string    | publiseringsstatus               |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED, PUBLISHED_METADATA, PUBLISHED_METADATA_AND_DOCUMENTS. **Publishing permission** required for anything else than NON_PUBLISHED                                                                                                             |

Reviewer: NON_PUBLISHED -> NOT_PUBLISHED
```suggestion
|           | string    | publiseringsstatus               |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED, PUBLISHED_METADATA, PUBLISHED_METADATA_AND_DOCUMENTS. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                             |
```

> |           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NON_PUBLISHED                                                                                                                                                    |

Reviewer: NON_PUBLISHED -> NOT_PUBLISHED
```suggestion
|           | string    | eInnsynStatus                    |    V     | Not published |                    |  |  V   |  V   |   V   |  V   |      |        | Allowed values: NOT_PUBLISHED. _Since 3.6.1_. **Publishing permission** required for anything else than NOT_PUBLISHED                                                                                                                                                    |
```

|           | ref       | refKryssreferanseFraRegistrering |          |               |                    |  |      |      |   V   |      |  V   |   V    | refers to **AbstraktRegistrering** that cross reference the current **AbstraktRegistrering**                                                                                                                                                                             |

### Moeteregistrering ([Basisregistrering](#basisregistrering-finalizable-object))
(Meeting Record)

Note: **EFO** required if finalized

Note: **EJO** required if journaled
| Field no. | Type      | Field                            | Not null |       Default      | Code list                |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:---------------------------------|:--------:|:------------------:|:-------------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
| M307      | string    | saksbehandler                    |    V     |                    |                          |  |  V   |  V   |   V   |  V   |      |        |         |

> |           | string    | saksnummer                       |          | registreringsIdent |                          |  |  V   |  V   |   V   |  V   |      |        |         |

Reviewer: `saksnummer` does not have a default value in Meeting Records:
```suggestion
|           | string    | saksnummer                       |          |   |                          |  |  V   |  V   |   V   |  V   |      |        |         |
```

|           | ref       | refTilknyttetAvskrivning          |          |             |                    |  |      |      |   V   |      |  V   |   V    |                                                                                                                                                                                                                                                                                                                                                                                                                 |
|           | ref       | refPresedens                      |          |             |                    |  |      |      |   V   |      |  V   |   V    | refers to **Presedens**                                                                                                                                                                                                                                                                                                                                                                                         |

> NOTE: journalansvarlig and journalansvarligBrukerIdent are inherited from the Registry Entry type and re-introduced in the table above because of the added constraints

Reviewer: ```suggestion
NOTE: journalansvarlig and journalansvarligBrukerIdent are inherited from the Basic Record type and re-introduced in the table above because of the added constraints.
```

|:----------|:----------|:-----------------------|:--------:|:-----------:|:----------:|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------------------------------------------------------------------|
| M665      | string    | flytFra                |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M665      | string    | flytFraBrukerIdent     |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M660      | string    | flytTil                |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |
| M660      | string    | flytTilBrukerIdent     |    V     |             |            |  |  V   |  V   |   V   |  V   |      |        |                                                                     |

> | M661      | timestamp | flytMottattDato        |    V     |             |   

Reviewer: `adresseKnr` should also be exactly 4 digits.

> | M661      | timestamp | flytMottattDato        |    V     |             |   

Reviewer: Typo: `allowd` -> `allowed`:
```suggestion
|           | string    | adressekode            |          |             |           |  |  V   |  V   |   V   |  V   |      |        | Only digits allowed                                                                                                             |
```

> | M661      | timestamp | flytMottattDato        |    V     |             |   

Reviewer: `bygningsnummer` can only contain digits too.

|           | ref       | refPrimaerKlassifikasjonssystem   |          |             |                |  |      |  V   |   V   |      |  V   |   V    | <p>Mandatory:</p><p>&bull; If null inherited from a child</p><p>&bull; If not null all childs must have the same primary KS </p>;                |
|           | ref       | refSekundaerKlassifikasjonssystem |          |             |                |  |      |      |   V   |      |  V   |   V    |                                                                                                                                                  |
|           | ref       | refMappe                          |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktMappe**</p>                                                                                                       |
|           | ref       | refRegistrering                   |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Basisregistrering**</p>                                                                                                   |

> |           | ref       | refArkivdelForloeper              |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p><p>&bull; Mandatory if arkivdelstatus is set to overlapping period</p> |

Reviewer: The comment here is incorrect, it is not mandatory to have the field set when arkivdelstatus is set to overlapping period. That is only true for the `refArkivdelArvtaker` field below.

|           | ref       | refArkivdelArvtaker               |          |             |                |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **Arkivdel**. Cross references are not allowed</p><p>&bull; Mandatory if arkivdelstatus is set to overlapping period</p> |

## Classification

### Klassifikasjonssystem ([Finalizable Object](#finalizable-object-basic-archival-object))
(Classification System)

> 

Reviewer: Here we also need the note for EFO requirements I think:
```
Note: **EFO** required if finalized
```

|           | ref       | refPresedens                     |          |                 |                    |  |      |      |   V   |      |  V   |   V    | refers to **Presedens**                                                                                                                                                                                                                                                                              |

### Saksmappe ([Abstract mappe](#abstract-mappe-finalizable-object))
(Case File)

Note: **EFO** required if finalized
| Field no. | Type      | Field                            | Not null |   Default   | Code list          |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:---------------------------------|:--------:|:-----------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
|           |           |                                  |          |             |                    |  |      |      |       |      |      |        |         |

### Part ([Basic Archival Object](#basic-archival-object))

> (Pary)

Reviewer: No need to split it into `Part` and `Sakspart`

|           | ref       | refPresedens                      |          |             |                    |  |      |      |   V   |      |  V   |   V    | refers to **Presedens**                                                                                                                                                                                                                                                                                                                                                                                         |

### Korrespondansepart ([Basic Archival Object](#basic-archival-object))
(Correspondance Party)

> 

Reviewer: ```suggestion
(Correspondence Party)
```

|           |           |                               |          |             |                        |  |      |      |       |      |      |        |                                                   |
|           | object    | virksomhetsspesifikkeMetadata |          |             |                        |  |  V   |  V   |       |      |      |        |                                                   |
|           |           |                               |          |             |                        |  |      |      |       |      |      |        |                                                   |
|           | ref       | refRegistrering               |    V     |             |                        |  |      |  V   |   V   |      |  V   |        | <p>&bull; references **AbstraktRegistrering**</p> |

### Dokumentflyt ([Basic Archival Object](#basic-archival-object))
(Document Flow)

| Field no. | Type      | Field                  | Not null |   Default   | Code list  |  | Save | View | Query | Sort | Link | Unlink | Comment                                                             |

> |:----------|:----------|:----------

Reviewer: Here we also need the note for EFO requirements I think:
```
Note: **EFO** required if finalized
```

### Arkivskaper ([Basic Archival Object](#basic-archival-object))
(Fonds Creator)

Note: **EFO** required if finalized
| Field no. | Type      | Field                  | Not null |   Default   | Code list |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:-----------------------|:--------:|:-----------:|:----------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
| M006      | string    | arkivskaperIdent       |    V     |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
| M023      | string    | arkivskaperNavn        |    V     |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
| M021      | string    | beskrivelse            |          |             |           |  |  V   |  V   |   V   |  V   |      |        |         |
|           |           |                        |          |             |           |  |      |      |       |      |      |        |         |
|           | ref       | refArkiv               |    V     |             |           |  |      |  V   |   V   |      |  V   |        | &nbsp;  |

> ### Arkivdel ([Basic Archival Object](#basic-archival-object))

Reviewer: Arkivdel is also finalizable.

|           | ref       | refMappeSomSekundaer          |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktMappe** as a secondary **Klasse**</p> <p>&bull; **refBarnKlasse** cannot be set if **refMappeSomSekundaer** is set</p>              |
|           | ref       | refRegistreringSomPrimaer     |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktRegistrering** as a primary **Klasse**</p> <p>&bull; **refBarnKlasse** cannot be set if **refRegistreringSomPrimaer** is set</p>    |
|           | ref       | refRegistreringSomSekundaer   |          |             |                   |  |      |      |   V   |      |  V   |   V    | <p>&bull; references **AbstraktRegistrering** as a secondary **Klasse**</p><p>&bull; **refBarnKlasse** cannot be set if **refRegistreringSomSekundaer** is set</p> |

## Folders

> ### File ([Finalizable Object](#finalizable-object-basic-archival-object))

Reviewer: I think this should become `Abstract mappe` and you should delete the `Abstract mappe` section which is a bit below.

Reviewer: Actually, the `Abstract mappe` section that is a bit below should probably be ranamed to `Saksmappe` and the common abstract fields there should just get removed.

Note: **EFO** required if finalized
| Field no. | Type      | Field                            | Not null |     Default     | Code list         |  | Save | View | Query | Sort | Link | Unlink | Comment                                                                                                                         |
|:----------|:----------|:---------------------------------|:--------:|:---------------:|:------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------------------------------------------------------------------------------------------------------------------------------|

> | M003      | string    | mappeIdent                       |          | Auto set to ID  |                   |  |  cw  |  V   |   V   |  V   |      |        | Default implemented in the descendent classes                                                                                   |

Reviewer: The file ID is auto set to the basic object ID for Folders and Meeting Folders, however for Case Files the file ID is auto set to \<case year\>/\<case sequence number\>.

|           | string    | mappetype                        |          |                 | mappetype         |  |  V   |  V   |   V   |      |      |        |                                                                                                                                 |
| M020      | string    | tittel                           |    V     |                 |                   |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                 |
| M025      | string    | offentligTittel                  |          |                 |                   |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                 |
| M021      | string    | beskrivelse                      |          |                 |                   |  |  V   |  V   |   V   |  V   |      |        |                                                                                                                                 |

> |           | timestamp | screeningEndDate                 |          |                 |                   |  |      |      |       |      |      |        |                                                                                                                                 |

Reviewer: We don't have that field in this web services.

| M088      | string    | moetesakstype                    |          |                    | moetesakstype            |  |  V   |  V   |   V   |      |      |        |         |
| M055      | string    | moeteregistreringsstatus         |          |                    | moeteregistreringsstatus |  |  V   |  V   |   V   |      |      |        |         |

### Arkivnotat ([Basisregistrering](#basisregistrering-finalizable-object))
(Archive Note)

_Since: 3.6.1_

Note: **EFO** required if finalized

Note: **EJO** required if journaled
| Field no. | Type      | Field                            | Not null |   Default   | Code list          |  | Save | View | Query | Sort | Link | Unlink | Comment |
|:----------|:----------|:---------------------------------|:--------:|:-----------:|:-------------------|:-|:----:|:----:|:-----:|:----:|:----:|:------:|:--------|
| M110      | date      | offentlighetsvurdertDato         |          |             |                    |  |  V   |  V   |   V   |  V   |      |        |         |

> ### Registry Entry ([Basisregistrering](#basisregistrering-finalizable-object))

Reviewer: You have the Registry Entry section here, which is not complete, and then a section for the Journalpost, which is complete. I think this Registry Entry section is redundant?

| M014      | number    | journalsekvensnummer              |    V     |      V      |                    |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |
| M015      | number    | journalpostnummer                 |    V     |      V      |                    |  |  cw  |  V   |   V   |  V   |      |        |                                                                                                                                                                                                                                                                                                                                                                                                                 |

> |           | string    | journalansvarlig                  |    V     |      V      |                    |  |  V   |  V   |   V   |  V   |      |        | <p>&bull; The name of the person responsible for this Journalpost (similar to M306 Saksmappe#saksansvarlig)</p>                                                                                                                                                                                                                                                                                                 |

Reviewer: `journalansvarlig` and `journalansvarligBrukerIdent` should be part of `Basisregistrering`.

Reviewer: Actually, we have the issue that the fields are necessary for Journalpost, but optional for all other record types :smile:

Author: Addressed in a separate commit: b41b3cb

|           | boolean   | skjermEInnsyn                 |          |    false    |                        |  |  V   |  V   |   V   |      |      |        | _Since 3.6.1_                                     |
|           |           |                               |          |             |                        |  |      |      |       |      |      |        |                                                   |
|           | object    | virksomhetsspesifikkeMetadata |          |             |                        |  |  V   |  V   |       |      |      |        |                                                   |
|           |           |                               |          |             |                        |  |      |      |       |      |      |        |                                                   |
|           | ref       | refRegistrering               |    V     |             |                        |  |      |  V   |   V   |      |  V   |        | <p>&bull; references **AbstraktRegistrering**</p> |

### Dokumentflyt ([Basic Archival Object](#basic-archival-object))
(Document Flow)

> 

Reviewer: Presedens is also finalizable.

> 

Reviewer: Missing a space:
```suggestion
### Dokument ([Finalizable Object](#finalizable-object-basic-archival-object))
```
